client/web/compose/src/views/Admin/Pages/Builder.vue

```diff
@@ -485,10 +502,28 @@ export default {
       return this.hasChildren || !this.page.canDeletePage || !!this.page.deletedAt
     },
 
-    selectableExistingBlocks () {
+    selectableExistingLayoutBlocks () {
       return this.page.blocks.filter(({ blockID }) => !this.usedBlocks.some(b => b.blockID === blockID))
     },
 
+    selectableNamespaceGlobalBlocks () {
+      const { namespaceID } = this.namespace
+      const namespace = this.namespaces.find((n) => n.namespaceID === namespaceID)
```

Comment:
I would add a method to the store that would get the namespace blocks and change the ID's like you do here.